0000-catalog.md

```diff
@@ -0,0 +1,789 @@
+- RFC PR:
+- Hyperledger Grid Issue: [GRID-95](https://jira.hyperledger.org/browse/GRID-95)
+
+# Summary
+
+[summary]: #summary
+
+This RFC proposes a generic implementation for a Hyperledger Grid _Product
+Catalog_. Product Catalogs will be represented by a unique identifier, which
+will be referenced by the _catalog_products_ included in the catalog. That
+catalog_id can be used to share a grouping of products that can be shared with 
+one or more organizations. The Grid Catalog will contain 4 fields. A 
+`catalog_id`, a `catalog_owner`, a `name` for the catalog, and a repeated 
+field of PropertyValues for custom fields.
+
+In addition, a base catalog product schema is included as a way to demonstrate
+enforcement of additional product properties. Any grid product added to a
+catalog will adhere to the "catalog_product" schema (or another custom-defined 
+schema). The example schema in particular will enforce a required `catalog_id`
+field, a required `status` field, an optional `return_policy`, and an 
+optional `price` field.
+
+# Motivation
+
+[motivation]: #motivation
+
+The Grid Catalog implementation is designed for sharing an assortment of
+products between organizations. Each product in a catalog will contain a number
+of master data elements. Using the Grid Catalog, organizations can share
+collections of related/unrelated products with other participants in their
+network.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+## Entities
+
+A `catalog` is a high-level construct that `catalog_products` can
+reference. A _catalog_product_ is an instance of a Grid Product with additional 
+schema defined attributes specific to the trading partner involved. The 
+"catalog_product" schema can be extended to include any number of attributes that 
+are required by an organization. A catalog can be referenced, or used to share 
+item level information in a supply chain. A catalog is referenced by a 
+"catalog_product" via its `catalog_id`. A catalog also has an `owner` (the 
+organization that creates the catalog). Also a `name` for the catalog.
+
+A catalog, like a product, can also have one or more `properties`. Properties
+are described in the Grid Primitives RFC. A _property namespace_ contains
+multiple _property schemas_. A property schema associates a name (such as
+"length") with a data type (such as integer). A catalog may want additional 
+properties to support executing custom smart contract, or trigger some type of 
+check. An example could be an expiry_date for an entire catalog. 
+
+## Transactions
+
+Catalogs are managed by submitting transactions to Hyperledger Grid, which will
+process them with the Grid Catalog smart contract. The following transactions
+are supported:
+
+**catalog actions:**
+
+- CatalogCreate - create a catalog and store it in state
+- CatalogUpdate - update a catalog in state
+- CatalogDelete - deletes a catalog from state and associated catalog_products
+
```

Comment:
Would `CatalogRead` be valid?

Reply:
I think a CatalogRead event could be handled at an application level. Would there be a compelling reason to include it in the protocol? Distributing the information to other nodes that someone has read something, keeping track of who has read what and when? 

Possibly, again it could also be handled at an application level.

Reply:
Chatted with James and Shawn on some additional detail. I agree with your response 👍